### PR TITLE
Test state additional tests

### LINKED ISSUE #610: Test state extensive testing
Cases to test:
- [ ] send_message_to_l1_syscall
- [ ] library_call_syscall
- [ ] block replace class deploy and get block hash turn off.
- [ ] sequencer address 
- [ ] Contract can't call test contract

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
Closes #610 

## Introduced changes

Cases to test:
- [x] library_call_syscall
- [x] get_block_hash_syscall
- [x] block replace class deploy and call turn off.
- [x] sequencer address 
- [x] Contract can't call test contract

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]

File: crates/forge/tests/integration/test_state.rs [hunk 1]
```diff
@@ -126,10 +136,215 @@ fn test_simple_syscalls() {
...
+                    use result::ResultTrait;
+                    use starknet::{ ClassHash, library_call_syscall, ContractAddress};
+
+                    #[storage]
+                    struct Storage {
+                    }
+
+                    #[external(v0)]
+                    fn get_sequencer_address(ref self: ContractState) -> ContractAddress {
+                        starknet::get_block_info().unbox().sequencer_address
```
Comment (cptartur) [inline crates/forge/tests/integration/test_state.rs:155]: Isn't this handled on the cheatnet side? I wonder if it makes more sense to have a test for that there.

File: crates/forge/tests/integration/test_state.rs [hunk 2]
```diff
@@ -126,10 +136,215 @@ fn test_simple_syscalls() {
...
+        #[test]
+        fn test_get_execution_info() {
+            let class_hash = declare('LibraryContract').class_hash;
+            let lib_dispatcher = ILibraryContractSafeLibraryDispatcher { class_hash };
+            let value = lib_dispatcher.get_value().unwrap();
+            assert(value == 0, 'Incorrect state');
+            lib_dispatcher.set_value(10);
+            let value = lib_dispatcher.get_value().unwrap();
+            assert(value == 10, 'Incorrect state');
+        }
```
Comment (cptartur) [inline crates/forge/tests/integration/test_state.rs:198]: I think we should somehow verify that the library calls really take place in context of the caller state (i.e. state of the called contract doesn't change).

File: crates/forge/tests/integration/test_state.rs (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply crates/forge/tests/integration/test_state.rs]: ? It tests consistency of values between cheatnet and tests

File: crates/forge/tests/integration/test_state.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/forge/tests/integration/test_state.rs:182]: > state of the called contract

what do you mean by called contract 🤔

Comment (MaksymilianDemitraszek) [reply crates/forge/tests/integration/test_state.rs:182]: If you elaborate and it makes sense I will still address it in a PR fixing library calls bcs there are still some problems and I don't want to mix in to many changes here https://github.com/foundry-rs/starknet-foundry/issues/697

Comment (MaksymilianDemitraszek) [reply crates/forge/tests/integration/test_state.rs:182]: discussed on slack, I wil try to come up with some test cases checking if the proper storage space is modified

**Pull request merged**